lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -615,6 +615,34 @@ def test_submission_history_contents(self):
         self.assertIn("Score: 3.0 / 3.0", response_content)
         self.assertIn('#4', response_content)
 
+    @ddt.data('America/New_York',  # UTC - 5
+              'Asia/Pyongyang',      # UTC + 9
+              'Europe/London',       # UTC
+              'Canada/Yukon',        # UTC - 8
+              'Europe/Moscow',)      # UTC + 3
+    def test_submission_history_timezone(self, timezone):
+        with (override_settings(TIME_ZONE=timezone)):
+            course = CourseFactory.create()
+            course_key = course.id
+            client = Client()
+            admin = AdminFactory.create()
+            client.login(username=admin.username, password='test')
+            state_client = DjangoXBlockUserStateClient(admin)
+            usage_key = course_key.make_usage_key('problem', 'test-history')
+            state_client.set(
+                username=admin.username,
+                block_key=usage_key,
+                state={'field_a': 'x', 'field_b': 'y'}
+            )
+            url = reverse('submission_history', kwargs={
+                'course_id': unicode(course_key),
+                'student_username': admin.username,
+                'location': unicode(usage_key),
+            })
+            response = client.get(url)
+            response_content = HTMLParser().unescape(response.content)
+            self.assertIn(settings.TIME_ZONE, response_content)
```

Comment:
This is a pretty weak test-- it just verifies that the time zone appears in the response content, but does not check that the date is properly converted to that timezone. If there is a way to mock out the updated date, then it would be possible to verify that the conversion is done.